core/lib/Thelia/Action/Product.php

```diff
@@ -100,6 +121,149 @@ public function create(ProductCreateEvent $event)
         $event->setProduct($product);
     }
 
+    /*******************
+     * CLONING PROCESS *
+     *******************/
+
+    /**
+     * @param ProductCloneEvent $event
+     */
+    public function cloneProduct(ProductCloneEvent $event)
+    {
+        // Get important datas
+        $lang = $event->getLang();
+        $originalProduct = $event->getOriginalProduct();
+        $dispatcher = $event->getDispatcher();
+
+        $originalProductDefaultI18n = ProductI18nQuery::create()
+            ->findPk([$originalProduct->getId(), $lang]);
+
+        $originalProductDefaultPrice = ProductPriceQuery::create()
+            ->findOneByProductSaleElementsId($originalProduct->getDefaultSaleElements()->getId());
+
+        // Cloning process
+
+        $clonedProduct = $this->createClone($originalProduct, $originalProductDefaultI18n, $originalProductDefaultPrice, $event);
+
+        $this->removeCloneDefaultPSE($clonedProduct, $originalProductDefaultPrice, $dispatcher);
+
+        $clonedProduct = $this->updateClone($clonedProduct, $originalProduct, $originalProductDefaultPrice, $dispatcher);
+
+        $this->cloneFeatureCombination($originalProduct->getId(), $clonedProduct->getId(), $dispatcher);
+
+        $this->cloneAssociatedContent($originalProduct->getId(), $clonedProduct, $dispatcher);
```

Comment:
You should trigger the FILE_CLONE and PSE_CLONE events from this method, and enclose the calls in the DB transaction.